src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions_2.cs

```diff
@@ -5,18 +5,19 @@ namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
     internal static partial class ISymbolExtensions
     {
-        public static bool IsValueParameter(this ISymbol symbol)
+        public static bool IsImplicitValueParameter(this ISymbol symbolOpt)
         {
-            if (symbol is IParameterSymbol)
+            if (symbolOpt is IParameterSymbol && symbolOpt.IsImplicitlyDeclared)
             {
-                var method = symbol.ContainingSymbol as IMethodSymbol;
+                var method = symbolOpt.ContainingSymbol as IMethodSymbol;
                 if (method != null)
                 {
                     if (method.MethodKind == MethodKind.EventAdd ||
                         method.MethodKind == MethodKind.EventRemove ||
                         method.MethodKind == MethodKind.PropertySet)
                     {
-                        return symbol.Name == "value";
+                        // the name is value in C#, and Value in VB
+                        return symbolOpt.Name == "value" || symbolOpt.Name == "Value";
```

Comment:
Just to verify, but this should never make a parameter in c# called "Value" get considered as an IsImplicitValueParameter right?